ipa-core/src/protocol/ipa_prf/malicious_security/verifier.rs

```diff
@@ -166,17 +173,32 @@ where
 #[cfg(all(test, unit_test))]
 mod test {
     use crate::{
-        ff::{Fp31, U128Conversions},
+        ff::{Fp31, PrimeField, U128Conversions},
         protocol::ipa_prf::malicious_security::{
             lagrange::{CanonicalLagrangeDenominator, LagrangeTable},
             verifier::{
-                compute_g_differences, compute_sum_share, interpolate_at_r, recurse_u_or_v,
+                compute_g_differences, compute_sum_share, interpolate_at_r,
                 recursively_compute_final_check,
             },
         },
         secret_sharing::SharedValue,
     };
 
+    /// This function compresses the `u_or_v` values and returns the next `u_or_v` values.
+    fn recurse_u_or_v<F: PrimeField, const λ: usize>(
```

Comment:
this was moved inside the test module just to avoid changing unit tests, just to avoid making too big of a change